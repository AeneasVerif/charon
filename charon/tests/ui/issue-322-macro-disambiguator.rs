macro_rules! transmute {
    () => {{
        struct AssertIsAsBytes;
        let _ = AssertIsAsBytes;
    }};
}

fn main() {
    transmute!();
    transmute!();
}
