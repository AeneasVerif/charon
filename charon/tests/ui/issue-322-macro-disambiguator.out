# Final LLBC before serialization:

struct test_crate::main::AssertIsAsBytes {}

struct test_crate::main::AssertIsAsBytes#1 {}

// Full name: test_crate::main
fn main()
{
    let @0: (); // return
    let @1: (); // anonymous local
    let @2: test_crate::main::AssertIsAsBytes; // anonymous local
    let @3: (); // anonymous local
    let @4: test_crate::main::AssertIsAsBytes#1; // anonymous local

    storage_live(@1)
    storage_live(@2)
    @2 := test_crate::main::AssertIsAsBytes {  }
    storage_dead(@2)
    storage_dead(@1)
    storage_live(@3)
    storage_live(@4)
    @4 := test_crate::main::AssertIsAsBytes#1 {  }
    storage_dead(@4)
    storage_dead(@3)
    @0 := ()
    @0 := ()
    return
}



